clients/src/main/java/org/apache/kafka/clients/tools/VerifiableProducer.java

```diff
@@ -150,8 +174,24 @@ public static VerifiableProducer createFromArgs(String[] args) {
             producerProps.put(ProducerConfig.ACKS_CONFIG, Integer.toString(res.getInt("acks")));
             // No producer retries
             producerProps.put("retries", "0");
+            
+            // Properties specified using the --producer-prop key=val format override
+            // other properties
+            List<String> userSpecifiedProducerProps = res.getList("userSpecifiedProducerProps");
+            for (String prop: userSpecifiedProducerProps) {
```

Comment:
Do we have code for this somewhere already. Looks like it should be reusable across a bunch of tools.